torch_geometric/utils/dropout.py

```diff
@@ -208,3 +216,88 @@ def dropout_edge(edge_index: Tensor, p: float = 0.5,
         edge_mask = edge_mask.nonzero().repeat((2, 1)).squeeze()
 
     return edge_index, edge_mask
+
+
+def dropout_path(edge_index: Tensor, r: float = 0.5, walks_per_node: int = 1,
+                 walk_length: int = 3, p: float = 1, q: float = 1,
+                 num_nodes: Optional[int] = None, is_sorted: bool = False,
+                 training: bool = True) -> Tuple[Tensor, Tensor]:
+    r"""Drops edges from the adjacency matrix :obj:`edge_index`
+    based on random walks. The source nodes to start random walks from are
+    sampled with probability :obj:`r` from a Bernoulli distribution.
+
+    The method returns (1) the retained :obj:`edge_index`, (2) the edge mask
+    indicating which edges were retained.
+
+    Args:
+        edge_index (LongTensor): The edge indices.
+        r (float, optional): The ratio of source nodes to start
```

Comment:
A bit sad that we can not call this `p`. How important are the `node2vec` parameters here in your opinion?